external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts

```diff
@@ -237,15 +243,23 @@ export type TraceEvent =
         type: TraceEventKind.OpenFrame,
         id: number,
         name: string,
-        fileHash: string
+        srcFileHash: string
+        bcodeFileHash: undefined | string,
         isNative: boolean,
         localsTypes: string[],
-        localsNames: string[],
+        localsNames: ILocalInfo[],
         paramValues: RuntimeValueType[]
-        optimizedLines: number[]
+        optimizedSrcLines: number[]
+        optimizedBcodeLines: undefined | number[]
     }
     | { type: TraceEventKind.CloseFrame, id: number }
-    | { type: TraceEventKind.Instruction, pc: number, loc: ILoc, kind: TraceInstructionKind }
+    | {
+        type: TraceEventKind.Instruction,
+        pc: number,
+        srcLoc: ILoc,
+        bcodeLoc: undefined | ILoc,
```

Comment:
Can't `srcLoc` also be undefined in the case where there is no source code? Or is this actually a sum type like this
```
enum Location {
  One { location: ILoc },
  Both { source_location: ILoc, bytecode_location: ILoc }
```

If so wondering if it would be good to reflect this in the structure of the data as opposed this which is a bit more loose I feel?

Reply:
The short answer is that at this point, it is assumed that source locations are always there, and the bytecode ones are optional. The idea is to implement and test the reverse in one of the following steps as this PR is already pretty big...

Reply:
A bit of follow on this. The reason for the current assumption is that when you execute tests with traces generated, you will now also get disassembly. The other direction (having either source or bytecode optional) will only matter once we hook this up to traced obtained from the replay tool, but I do plan to work on this pretty much right away